src/m1_facilitator/Config/Config.ts

```diff
@@ -0,0 +1,189 @@
+// Copyright 2020 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import fs from 'fs-extra';
+import yaml from 'js-yaml';
+
+/**
+ * The class holds database configurations.
+ */
+export class DBConfig {
+  /** Database path */
+  public path: string;
+
+  public constructor() {
+    this.path = '';
+  }
+}
+
+/**
+ * It holds chain information like .
+ */
+export class Chain {
+  /** Chain rpc endpoint. */
+  public readonly nodeRpcEndpoint: string;
+
+  /** Subgraph web socket endpoint */
+  public readonly subGraphWsEndpoint: string;
+
+  /** Subgraph rpc endpoint */
+  public readonly subGraphRpcEndpoint: string;
+
+  /** Worker address. */
+  public readonly worker: string;
+
+  /** Worker password. */
+  private readonly _password?: string;
+
+  public constructor(
+    nodeRpcEndpoint: string,
+    subGraphWsEndpoint: string,
+    subGraphRpcEndpoint: string,
+    worker: string,
+    password?: string,
+  ) {
+    this.nodeRpcEndpoint = nodeRpcEndpoint;
+    this.subGraphWsEndpoint = subGraphWsEndpoint;
+    this.subGraphRpcEndpoint = subGraphRpcEndpoint;
+    this.worker = worker;
+    this._password = password;
+  }
+
+  /**
+   * Get the password for unlocking worker.
+   */
+  public get password(): string | undefined {
+    return this._password;
+  }
+}
+
+/**
+ * Represent metachain which contains origin and auxiliary chain objects.
+ */
+export class Metachain {
+  public readonly originChain: Chain;
+
+  public readonly auxChain: Chain;
+
+  public constructor(originChain: Chain, auxChain: Chain) {
+    this.originChain = originChain;
+    this.auxChain = auxChain;
+  }
+}
+
+/**
+ * The object represents facilitator manifest values.
+ */
+export default class Config {
+  public readonly version: string;
+
+  public readonly architectureLayout: string;
+
+  public readonly personas: string[];
+
+  public readonly chain: string;
+
+  public readonly metachain: Metachain;
+
+  public readonly dbConfig: DBConfig;
+
+  public readonly encryptedAccounts: Record<string, any>;
```

Comment:
Why `any`?

Reply:
This should be well defined abstraction.

Reply:
Updated with `Account` class abstraction